src/matlab2tikz.m

```diff
@@ -3424,6 +3424,16 @@ case size(cData,2) % specific y-data is given
     end
 end
 % ==============================================================================
+function BarWidth=getCorrectBarWidth(h)
+XData_bar=get(h,'XData');
+length_bar = length(XData_bar);
+BarWidth= get(h, 'BarWidth');
+if length_bar > 1
+    BarWidth = min(diff(XData_bar))*BarWidth;
```

Comment:
I am wondering why you use the `min`imal difference. Would `mean` make more sense?